cvmfs/telemetry_aggregator_influx.cc

```diff
@@ -0,0 +1,245 @@
+  int params = 0;
+
+  if (options_mgr->GetValue("CVMFS_INFLUX_HOST", &influx_host_)) {
+    if (influx_host_.size() > 1) {
+      params++;
+    } else {
+      LogCvmfs(kLogTelemetry, kLogDebug | kLogSyslogWarn,
+                                        "No value given for CVMFS_INFLUX_HOST");
+    }
+  }
+
+  std::string opt;
+  if (options_mgr->GetValue("CVMFS_INFLUX_PORT", &opt)) {
+      influx_port_ = static_cast<int>(String2Int64(opt.c_str()));
+      if (influx_port_ > 0 && influx_port_ < 65536) {
+        params++;
+      } else {
+        LogCvmfs(kLogTelemetry, kLogDebug | kLogSyslogWarn,
+                      "Invalid value for CVMFS_INFLUX_PORT [%s]", opt.c_str());
+      }
+  }
+
+  if (options_mgr->GetValue("CVMFS_INFLUX_METRIC_NAME", &influx_metric_name_))
+  {
+      params++;
+  }
+
+  if (!options_mgr->GetValue("CVMFS_INFLUX_EXTRA_TAGS", &influx_extra_tags_)) {
+    influx_extra_tags_ = "";
+  }
+
+  if (!options_mgr->GetValue("CVMFS_INFLUX_EXTRA_FIELDS",
+                              &influx_extra_fields_)) {
+    influx_extra_fields_ = "";
+  }
+
+  if (params == 3) {
+    is_zombie_ = false;
+    LogCvmfs(kLogTelemetry, kLogDebug, "Enabling influx metrics. "
+                                    "Send to [%s:%d] metric [%s]. Extra tags"
+                                    "[%s]. Extra fields [%s].",
+                                    influx_host_.c_str(),
+                                    influx_port_,
+                                    influx_metric_name_.c_str(),
+                                    influx_extra_tags_.c_str(),
+                                    influx_extra_fields_.c_str());
+  } else {
+    is_zombie_ = true;
+    LogCvmfs(kLogTelemetry, kLogDebug | kLogSyslogWarn,
+             "Not enabling influx metrics. Not all mandatory variables set: "
+             "CVMFS_INFLUX_METRIC_NAME, CVMFS_INFLUX_HOST, CVMFS_INFLUX_PORT");
+  }
+}
+
+/**
+ * Influx dataformat
+ * ( https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/ )
+ * Syntax
+   <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]
+ *
+ * Example
+   myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000
+*/
+std::string TelemetryAggregatorInflux::MakePayload() {
+  // measurement and tags
+  std::string ret = "" + influx_metric_name_ + "_absolute,repo=" + fqrn_;
+
+  if (influx_extra_tags_ != "") {
+    ret += "," + influx_extra_tags_;
+  }
+
+  // fields
+  ret += " ";
+  const int64_t revision = counters_["catalog_revision"];
+  ret += "catalog_revision=" + StringifyUint(revision);
+
+  std::string tok = ",";
+  for (std::map<std::string, int64_t>::iterator it
+      = counters_.begin(), iEnd = counters_.end(); it != iEnd; it++) {
+    if (it->second != 0) {
+      ret += tok + it->first + "=" + StringifyInt(it->second);
+
+      tok = ",";
+    }
+  }
+  if (influx_extra_fields_ != "") {
```

Comment:
If we don't have extra fields, the field list ends with a trailing comma. Is this ok?

Reply:
no it isnt. might have confused you with `tok = ",";` which is unneccesary.
it is `ret += tok + it->first + "=" + StringifyInt(it->second);` -- so comma is only added if there is a field following